include/nlohmann/detail/conversions/from_json.hpp

```diff
@@ -43,6 +43,24 @@ inline void from_json(const BasicJsonType& j, typename std::nullptr_t& n)
     n = nullptr;
 }
 
+template <typename BasicJsonType, typename ArithmeticTypeTarget, typename ArithmeticTypeSource>
+ArithmeticTypeTarget static_cast_check_range(const BasicJsonType& j)
+{
+    const auto val = *j.template get_ptr<ArithmeticTypeSource*>();
+    const auto min = std::numeric_limits<ArithmeticTypeTarget>::min();
+    const auto max = std::numeric_limits<ArithmeticTypeTarget>::max();
+    if (val < min && val > max)
```

Comment:
This `if` will never be true, unless you get some weird results from val and min/max having different types.  You want `||` not `&&`.  You probably also want `min` and `max` to have the same type as `val`.  You would also only need to do this test if `ArithmeticTypeTarget` and `ArithmeticTypeSource` are differen types.

It might also be better to do this as a pre-test that just throws on out of range, and leave the current lines alone for the actual assignments.